gdal/frmts/nitf/nitfdes.c

```diff
@@ -590,3 +590,98 @@ int NITFDESExtractShapefile(NITFDES* psDES, const char* pszRadixFileName)
 
     return TRUE;
 }
+
+/************************************************************************/
+/*                              NITFDESGetXml()                         */
+/************************************************************************/
+
+CPLXMLNode* NITFDESGetXml(NITFFile* psFile, int iSegment)
+{
+    CPLXMLNode* psDesNode = CPLCreateXMLNode(NULL, CXT_Element, "des");
+    NITFDES* psDes = NITFDESAccess(psFile, iSegment);
+    char** papszTmp = psDes->papszMetadata;
```

Comment:
Dereferencing psDes should be moved after testing it for NULL. 
(Note: if having this file as a .cpp one makes your life easier, you can do it, although I suspect it might cause initial warnings to solve. Using .c is legacy)
You could simplify a bit the 2 below error branches by assigning psDesNode to CPLCreateXMLNode() after

Reply:
Done.  Good to know on C v. CPP.  I'm not too keen to switch over now that it's mostly implemented, but if it's helpful to the project I could make a separate PR to get it building as CPP - let me know.

Reply:
> but if it's helpful to the project I could make a separate PR to get it building as CPP - let me know.

Was just informing that it could be a possibility if dealing with pure C was too painful